compiler/testSuite/14-bug/record-fv.bal

```diff
@@ -0,0 +1,16 @@
+import ballerina/io;
+
+type A record {|
+    string a;
+|};
+
+type B record {|
+    int a;
+|};
+
+type C A|B;
+
+public function main() {
+    C? b = {a: 1};
```

Comment:
Why do you think this is correct? Please quote relevant part of spec.

Reply:
For this and the [fallowing](https://github.com/ballerina-platform/nballerina/pull/999#discussion_r862556737),  issue I had in the spec was with the phrase "it is a compile-time error if this does not leave a single ... type descriptor" (relevant sections [mapping-constructor-expr](https://ballerina.io/spec/lang/2022R1/#mapping-constructor-expr), [list-constructor-expr](https://ballerina.io/spec/lang/2022R1/#list-constructor-expr). If this means 

1. If the number of type descriptors left is not equal to 1 (then it is a compile time error). Then we don't have a bug. JBallerina has one.
2. If the number of type descriptors left is less than 1 ie. 0 (then it is a compile time error). Then we have a bug and JBallerina is correct.

Reply:
It certainly means (1) not (2).  You need exactly one type descriptor to use as the inherent type. That's not the issue. The issue is whether anydata|anydata counts as a single type descriptor or two type descriptors.  I think the answer has to be two, because the spec does not define any equality relationship for type descriptors. So I don't think this is a bug in nballerina.